src/Pool.sol

```diff
@@ -773,6 +791,75 @@ abstract contract Pool {
         return subSenders[senderAddr][subSenderId].receiversHash;
     }
 
+    /// @notice Sets a new list of drips receivers of a user
+    /// @param userAddr The user address
+    /// @param currReceivers The list of the user's drips receivers which is currently in use.
+    /// If this function is called for the first time for the user, should be an empty array.
+    /// @param newReceivers The new list of the user's drips receivers.
+    /// Must be sorted by the drips receivers' addresses, deduplicated and without 0 weights.
+    /// Each drips receiver will be getting `weight / TOTAL_DRIPS_WEIGHTS`
+    /// share of the funds collected by the user.
+    function _setDripsReceiversInternal(
+        address userAddr,
+        DripsReceiver[] calldata currReceivers,
+        DripsReceiver[] calldata newReceivers
+    ) internal {
+        _assertCurrDripsReceivers(userAddr, currReceivers);
+        _assertDripsReceiversValid(newReceivers);
+        dripsReceivers[userAddr] = hashDripsReceivers(newReceivers);
+        emit DripsReceiversUpdated(userAddr, newReceivers);
+    }
+
+    /// @notice Validates a list of drips receivers
+    /// @param receivers The list of drips receivers
+    /// Must be sorted by the drips receivers' addresses, deduplicated and without 0 weights.
+    function _assertDripsReceiversValid(DripsReceiver[] calldata receivers) internal pure {
+        require(receivers.length <= MAX_DRIPS_RECEIVERS, "Too many drips receivers");
+        uint64 totalWeight = 0;
+        for (uint256 i = 0; i < receivers.length; i++) {
+            require(receivers[i].weight != 0, "Drips receiver weight is zero");
+            totalWeight += receivers[i].weight;
+            if (i > 0) {
+                address prevReceiver = receivers[i - 1].receiver;
+                address currReceiver = receivers[i].receiver;
+                require(prevReceiver <= currReceiver, "Drips receivers not sorted by address");
+                require(prevReceiver != currReceiver, "Duplicate drips receivers");
+            }
```

Comment:
Why are potential duplicates a problem? I can't see any major downside.

Reply:
That's a good point, I haven't thought about it! Unlike the sender receivers the drips receivers don't have to be consumed in a structured fashion. They could be zeroed, unsorted and duplicated, they'll still work. The only downside would be that it would generate unnecessary gas cost to have unnecessary entries and that it would emit noisy events. If we want to ban duplicates, we need to enforce sorting too, these requirements are bundle. If we don't enforce being sorted and stripped of zeroed entries, it may be also confusing that two equivalent lists of receivers can't be used as substitutes, because they generate different hashes.

What do you think? Should we keep the API strict or should we loosen it?

Reply:
In general i agree with strictness but in this case it costs gas.
However, we anyway need to iterate the list to check if `totalWeight` is smaller equal than `TOTAL_DRIPS_WEIGHTS`.

This we obviously can't skip.

Reply:
Therefore, we can keep it to ensure the order.